api/gateway-operator/v2alpha1/controlplane_types.go

```diff
@@ -352,6 +357,28 @@ type ControlPlaneConfigDump struct {
 	DumpSensitive ConfigDumpState `json:"dumpSensitive"`
 }
 
+// ControlPlaneObjectFilters defines filters to limit watched objects by the controllers.
+type ControlPlaneObjectFilters struct {
+	// Secrets defines the filters for watched secrets.
+	//
+	// +optional
+	Secrets *ControlPlaneFilterForSingleObject `json:"secrets,omitempty"`
+	// ConfigMaps defines the filters for watched config maps.
+	//
+	// +optional
+	ConfigMaps *ControlPlaneFilterForSingleObject `json:"configMaps,omitempty"`
+}
+
+// ControlPlaneFilterForSingleObject defines the filters for a certain type of object.
+type ControlPlaneFilterForSingleObject struct {
+	// AllowedLabel specifies a label that an object must have the label with value "true" to get watched by the controllers.
+	// For example, if the secrets.allowedLabel is set to "some-label", only secrets with label "some-label=true" are reconciled.
+	//
+	// +optional
+	// +kubebuilder:validation:MinLength=1
+	AllowedLabel *string `json:"allowedLabel,omitempty"` // REVIEW: add other validation of label keys, like the prefix/name format, max length, used chars?
```

Comment:
IMHO allowed doesn't have the accurate meaning for this use case, WDYT about using:

```suggestion
	// MustHaveLabel specifies a label that an object must have set to "true" to be watched by the controllers.
	// For example, if the secrets.mustHaveLabel is set to "some-label", only secrets with label "some-label=true" are reconciled.
	//
	// +optional
	// +kubebuilder:validation:MinLength=1
	MustHaveLabel *string `json:"mustHaveLabel,omitempty"`
```

>  // REVIEW: add other validation of label keys, like the prefix/name format, max length, used chars?

I think we can solve that when there's user/product demand for this.